app/concepts/matestack/ui/core/app/app.js

```diff
@@ -16,11 +17,11 @@ const componentDef = {
   mounted: function(){
     const self = this;
     window.onpopstate = (event) => {
-      let needToNavigate = self.currentPathName !== document.location.pathname ||
-        self.currentOrigin !== document.location.origin ||
-        self.currentSearch !== document.location.search
-
-      if (needToNavigate){
+      if (isNavigatingToAnotherPage({
+          origin: self.currentOrigin,
+          pathName: self.currentPathName,
+          search: self.currentSearch
```

Comment:
Small thing, having all these "flat" in the store seemed weird to me before. This makes me feel it's even weirder. How about storing them in an object in the store? Like: `currentLocation: {all_da_values}` then that object could be passed on as a whole. It also makes it clearer that all these values belong together and are not of too much use on their own.